src/Model/Behavior/VersionBehavior.php

```diff
@@ -428,4 +434,29 @@ protected function _referenceName()
 
         return $name;
     }
+
+    /**
+     * Converts fields to the appropriate type to be stored in the version, and
+     * to be converted from the version record to the entity
+     *
+     * @param array $fields Fields to convert
+     * @param string $direction Direction (toPHP or toDatabase)
+     * @return array
+     */
+    protected function _convertFieldsToType(array $fields, $direction)
```

Comment:
Can we get a test for this?

Reply:
It is implicitly tested in the new tests, except for the invalid argument exception which I can add a test for.

Reply:
I want an explicit test.